src/script/common/helper.cpp

```diff
@@ -77,3 +78,31 @@ std::string LuaHelper::readParam(lua_State *L, int index)
 	auto sv = readParam<std::string_view>(L, index);
 	return std::string(sv); // a copy
 }
+
+double LuaHelper::my_lua_string_to_double(lua_State *L, std::string_view sv)
+{
+	if (sv == "nan")
+		return std::numeric_limits<double>::quiet_NaN();
+	if (sv == "inf")
+		return std::numeric_limits<double>::infinity();
+	if (sv == "-inf")
+		return -std::numeric_limits<double>::infinity();
+	lua_pushlstring(L, sv.data(), sv.size());
+	return lua_tonumber(L, -1);
```

Comment:
I was thinking this method should leave the value on the stack instead.
consider that the only current usage is equivalent to this:
```c
// (some rarely used paths here)
lua_pushlstring(L, sv.data(), sv.size());
double n = lua_tonumber(L, -1);
lua_pushnumber(L, n);
```